src/engine.rs

```diff
@@ -0,0 +1,108 @@
+use anyhow::{anyhow, Result};
+use std::{
+    path::PathBuf,
+    time::{Duration, Instant},
+};
+use wasmtime::{Engine, Linker, Module, Store};
+
+use wasmtime_wasi::sync::WasiCtxBuilder;
+
+use crate::run_statistics::RunStatistics;
+
+pub fn run(script_path: PathBuf, input_path: PathBuf) -> Result<RunStatistics> {
+    let engine = Engine::default();
+    let module = Module::from_file(&engine, &script_path)
+        .map_err(|e| anyhow!("Couldn't load script {:?}: {}", &script_path, e))?;
+
+    let input: serde_json::Value = serde_json::from_reader(
+        std::fs::File::open(&input_path)
+            .map_err(|e| anyhow!("Couldn't load input {:?}: {}", &input_path, e))?,
+    )
+    .map_err(|e| anyhow!("Couldn't load input {:?}: {}", &input_path, e))?;
+    let input = serde_json::to_vec(&input)?;
+
+    let input_stream = wasi_common::pipe::ReadPipe::new(std::io::Cursor::new(input));
+    let output_stream = wasi_common::pipe::WritePipe::new_in_memory();
+    let error_stream = wasi_common::pipe::WritePipe::new_in_memory();
+
+    let runtime: Duration;
+
+    {
+        // Link WASI and construct the store.
+        let mut linker = Linker::new(&engine);
+        wasmtime_wasi::add_to_linker(&mut linker, |s| s)?;
+        let wasi = WasiCtxBuilder::new()
+            .stdin(Box::new(input_stream))
+            .stdout(Box::new(output_stream.clone()))
+            .stderr(Box::new(error_stream.clone()))
+            .inherit_args()?
+            .build();
+        let mut store = Store::new(&engine, wasi);
+
+        linker.module(&mut store, "", &module)?;
+
+        let start = Instant::now();
+
+        // Execute the module
```

Comment:
Nit: I think you could alter the code below to make this clear without the comment:
```
let module_result = ...
```